Snowflake.Data/Core/Session/SFSession.cs

```diff
@@ -460,20 +464,30 @@ internal RequestQueryContext GetQueryContextRequest()
             return _queryContextCache.GetQueryContextRequest();
         }
 
-        internal void UpdateDatabaseAndSchema(string databaseName, string schemaName)
+        internal void UpdateSessionProperties(QueryExecResponseData responseData)
         {
-            // with HTAP session metadata removal database/schema
-            // might be not returened in query result
-            if (!String.IsNullOrEmpty(databaseName))
-            {
-                this.database = databaseName;
-            }
-            if (!String.IsNullOrEmpty(schemaName))
+            // with HTAP session metadata removal database/schema might be not returned in query result
+            UpdateSessionProperty(ref database, responseData.finalDatabaseName);
+            UpdateSessionProperty(ref schema, responseData.finalSchemaName);
+            UpdateSessionProperty(ref role, responseData.finalRoleName);
+            UpdateSessionProperty(ref warehouse, responseData.finalWarehouseName);
+        }
+
+        private void UpdateSessionProperty(ref string initialSessionValue, string finalSessionValue)
+        {
+            // with HTAP session metadata removal database/schema might be not returned in query result
+            if (!String.IsNullOrEmpty(finalSessionValue))
             {
-                this.schema = schemaName;
+                if (!String.IsNullOrEmpty(initialSessionValue) && initialSessionValue != finalSessionValue)

```

Comment:
string lowercase, you should respect quoted values strictly but ignore case for unquoted values.